jena-arq/src/main/java/org/apache/jena/http/HttpLib.java

```diff
@@ -0,0 +1,711 @@
+     * @param httpRequest
+     * @param bodyHandler
+     * @return HttpResponse
+     */
+    public static <T> HttpResponse<T> executeJDK(HttpClient httpClient, HttpRequest httpRequest, BodyHandler<T> bodyHandler) {
+        try {
+            // This is the one place all HTTP requests go through.
+            logRequest(httpRequest);
+            HttpResponse<T> httpResponse = httpClient.send(httpRequest, bodyHandler);
+            logResponse(httpResponse);
+            return httpResponse;
+        //} catch (HttpTimeoutException ex) {
+        } catch (IOException | InterruptedException ex) {
+            if ( ex.getMessage() != null ) {
+                // This is silly.
+                // Rather than an HTTP exception, bad authentication becomes IOException("too many authentication attempts");
+                // or IOException("No credentials provided") if the authenticator decides to return null.
+                if ( ex.getMessage().contains("too many authentication attempts") ||
+                     ex.getMessage().contains("No credentials provided") ) {
+                    throw new HttpException(401, HttpSC.getMessage(401));
+                }
+            }
+            throw new HttpException(httpRequest.method()+" "+httpRequest.uri().toString(), ex);
+        }
+    }
+
+    /*package*/ static CompletableFuture<HttpResponse<InputStream>> asyncExecute(HttpClient httpClient, HttpRequest httpRequest) {
+        logAsyncRequest(httpRequest);
+        return httpClient.sendAsync(httpRequest, BodyHandlers.ofInputStream());
+    }
+
+    /** Push data. POST, PUT, PATCH request with no response body data. */
+    public static void httpPushData(HttpClient httpClient, Push style, String url, Consumer<HttpRequest.Builder> modifier, BodyPublisher body) {
+        HttpResponse<InputStream> response = httpPushWithResponse(httpClient, style, url, modifier, body);
+        handleResponseNoBody(response);
+    }
+
+    // Worker
+    /*package*/ static HttpResponse<InputStream> httpPushWithResponse(HttpClient httpClient, Push style, String url,
+                                                                      Consumer<HttpRequest.Builder> modifier, BodyPublisher body) {
+        URI uri = toRequestURI(url);
+        HttpRequest.Builder builder = requestBuilderFor(url);
+        builder.uri(uri);
+        builder.method(style.method(), body);
+        if ( modifier != null )
+            modifier.accept(builder);
+        HttpResponse<InputStream> response = execute(httpClient, builder.build());
+        return response;
+    }
+
+
+    /** Request */
+    private static void logRequest(HttpRequest httpRequest) {
+        // Uses the SystemLogger which defaults to JUL.
+        // Add org.apache.jena.logging:log4j-jpl
+        // (java11 : 11.0.9, if using log4j-jpl, logging prints the request as {0} but response OK)
+//        httpRequest.uri();
+//        httpRequest.method();
+//        httpRequest.headers();
```

Comment:
Yes - For later. At least the logging calls are in-place and we can put in logging.

It may not be necessary and it can just be turn on the JDK logging but the line 599 is about a problem I found where the JDFK code uses `{0}` which is assuming java.util.logging.